File: lc66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int length=digits.size();
        for(int i=length;i!=0;i--){
            if(digits[i-1]!=9) {
                digits[i-1]++;
                break;
            }
            else{
                digits[i-1]=0;
                if(i==1){
                    
                    digits.insert(digits.begin(),1);
                }
                
                
            }
            
        }
        return digits;
    }
};

File: lc66. Plus One.py
class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        return [int(s) for s in list(str(int(''.join([str(x) for x in digits])) + 1))]


digits = [4, 3, 2, 1]
s = Solution()
print(s.plusOne(digits))
